cmd/soroban-rpc/internal/db/migration.go

```diff
@@ -122,32 +128,20 @@ type guardedMigration struct {
 func newGuardedDataMigration(
 	ctx context.Context, uniqueMigrationName string, logger *log.Entry, factory migrationApplierFactory, db *DB,
 ) (Migration, error) {
-	migrationDB := &DB{
-		cache:            db.cache,
-		SessionInterface: db.SessionInterface.Clone(),
-	}
-	if err := migrationDB.Begin(ctx); err != nil {
-		return nil, err
-	}
 	metaKey := "Migration" + uniqueMigrationName + "Done"
-	previouslyMigrated, err := getMetaBool(ctx, migrationDB, metaKey)
+	previouslyMigrated, err := getMetaBool(ctx, db, metaKey)
 	if err != nil && !errors.Is(err, ErrEmptyDB) {
-		err = errors.Join(err, migrationDB.Rollback())
+		err = errors.Join(err, db.Rollback())
```

Comment:
it seems strange that the migration is responsible for rolling back the transaction but it did not create the transaction. I think it would be more appropriate for the daemon to handle rolling back since it created the transaction in the first place

Reply:
@tamirms Yeah this code was being used in the previous version where tx was being created in `guardedMigration`. It does not need to be here now.